### PR TITLE
feat: add health package

### PR DESCRIPTION
**Pull request by kentquirk**
## Which problem is this PR solving?

- Adds an internal health monitor package that will be used for health checks

## Short description of the changes

- Add health object
- Add tests for it

### CODE REVIEW

File: internal/health/health.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+// TickerTime is the interval at which we will check the health of the system.
+var TickerTime = 1 * time.Second
+
+// The Health object is the main object that services will interact with.
+// When services are registered, they will be expected to report in at least once every timeout interval.
+// If they don't, they will be marked as not alive.
+type Health struct {
+	Clock    clockwork.Clock
```
Comment (VinozzZ) [inline internal/health/health.go:48]: Do we want to inject this service?

File: internal/health/health.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	return nil
+}
+
+func (h *Health) Stop() error {
+	close(h.done)
+	return nil
+}
+
+func (h *Health) ticker() {
+	fmt.Println("starting ticker")
```
Comment (VinozzZ) [inline internal/health/health.go:75]: do we still need the print lines here?

File: internal/health/health.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	h.mut.RLock()
+	defer h.mut.RUnlock()
+	// if any counter is not positive, we're not ready
+	for _, a := range h.timeLeft {
+		if a <= 0 {
+			return false
+		}
+	}
+
+	// if no one has registered yet, we're not ready
```
Comment (VinozzZ) [inline internal/health/health.go:171]: If no one has registered, then `h.timeLeft` will be all `0` and that should be covered by line 166?

**Review by VinozzZ — APPROVED** [review]

File: internal/health/health.go (hunk 3 above)
Comment (kentquirk) [reply internal/health/health.go:171]: If no one has registered, these loops don't get run, so line 166 never gets executed.

File: internal/health/health.go (hunk 2 above)
Comment (kentquirk) [reply internal/health/health.go]: oops

File: internal/health/health.go (hunk 1 above)
Comment (kentquirk) [reply internal/health/health.go]: Yes!

**Pull request merged**